easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -589,11 +604,14 @@ def _parse_dependency(self, dep, hidden=False):
             'external_module_metadata': {},
         }
         if isinstance(dep, dict):
-            dependency.update(dep)
             # make sure 'dummy' key is handled appropriately
             if 'dummy' in dep and not 'toolchain' in dep:
                 dependency['toolchain'] = dep['dummy']
 
+            dependency.update(dep)
```

Comment:
hmm, moving this down may change the value for the `toolchain` key, if it's defined in `dep`?